dataset.py

```diff
@@ -1,30 +1,77 @@
 import os
-from torch.utils.data import Dataset
+import pandas as pd
+from torch.utils.data import Dataset, DataLoader
 from PIL import Image
-
+from torchvision import transforms
 class Flickr8kDataset(Dataset):
 
-	def __init__(self, dataset_folder='./dataset'):
+	def __init__(self, dataset_folder='./dataset',transform=None):
+		super().__init__()
+		self.transform = transform
 		self.images_folder = os.path.join(dataset_folder,'images')
-		self.captions_file = open(os.path.join(dataset_folder,'captions.txt'),'r')
-		self.dataset = [line.strip() for line in self.captions_file.readlines()[1:]]
-		self.captions_file.close()
+		self.dataframe = pd.read_csv(open(os.path.join(dataset_folder,'captions.txt'),'r'))
 
 	def read_image(self,filename):
+		"""
+		Reads the image file based on the filename
+
+		Args:
+			filename (string): Filename that contains the image
+
+		Returns:
+			[Image]: Returns a PIL.Image object 
+		"""
 		image_path = os.path.join(self.images_folder, filename)
 		image = Image.open(image_path)
 		return image
 
 	def __len__(self):
-		return len(self.dataset)
+		return len(self.dataframe)
 
 	def __getitem__(self,idx):
-		filename, caption = self.dataset[idx].split(',')
+		"""
+		Process the dataset according to our needs.
+		The image needs to be resize and normalize.
+
+		Args:
+			idx (int): Index of the dataset element to be returned
+
+		Returns:
+			torch.Tensor: 	Processed image
+			string:			Caption of the image
+		"""
+		filename, caption = self.dataframe.iloc[idx,:]
 		image = self.read_image(filename)
-		label = caption
-		return image,label
+
+		if self.transform:
+			image = self.transform(image)
+
+		return image, caption
 
 
 if __name__ == "__main__":
-	dataset = Flickr8kDataset()
-	dataset.__getitem__(15)
\ No newline at end of file
+	image_size = (128,128)
+	batch_size = 128
+	transform = transforms.Compose([
+		transforms.ToTensor(),
+		transforms.Resize(image_size),
+		# The normalize parameters depends on the model we're gonna use
+		# If we apply transfer learning from a model that used ImageNet, then
+		# we should use the ImageNet values to normalize the dataset.
+		# Otherwise we could just normalize the values between -1 and 1 using the 
+		# standard mean and standard deviation
+		# transforms.Normalize(mean=[0.485, 0.456, 0.406],std=[0.229, 0.224, 0.225]),
+		transforms.Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5]),
+	])
+	dataset = Flickr8kDataset(transform=transform)
+
+	# Test the dataloader
+	dataloader = DataLoader(dataset=dataset,batch_size=batch_size)
+	batch_sample = next(iter(dataloader))
+	images, captions = batch_sample
+
+	assert images.size() == (batch_size,) + (3,) + image_size
```

Comment:
I've added a couple of simple asserts to check that everything works as expected